src/components/Tablecell.tsx

```diff
@@ -0,0 +1,140 @@
+		symbol
+	}
+}
+
+function Cell(props: TableCellProps & { insideModal?: boolean }) {
+
+	switch (props.type) {
+		case "society":
+			return <>
+					<div>
+                        <GridItem className={`society block-${props.society.block.toLowerCase()}`}>
+                            <span className="index">{props.index}</span>
+                            <span className="year">{props.society.year}</span>
+                            <span className="recognized">{props.society.tsg_recognized ? "*" : ""}</span>
+                            <span className="symbol">{props.society.symbol}</span>
+                            <span className="size">{props.society.size}</span>
+                        </GridItem>
+					</div>
+				</>;
+		case "empty":
+			return <GridItem className="society empty-cell"></GridItem>;
+		case "inner-transition":
+			return <GridItem className="society inner-transition-cell">
+				<span className="symbol">↓</span>
+			</GridItem>;
+		case "the-unknown-soc":
+			return <>
+				<div data-tooltip-id="the-unknown-soc" data-tooltip-delay-hide={0}>
+					<GridItem className={`society the-unknown-soc`} >
+						<span className="index">?</span>
+						<span className="year">?</span>
+						<span className="symbol">Us</span>
+						<span className="size">?</span>
+					</GridItem>
+				</div>
+				<Tooltip className="soc-tooltip" id="the-unknown-soc" delayHide={0} delayShow={0}>
+					<span className="soc-tooltip-name">The Unknown Society</span>
+					<p className="soc-toolip-desc">Incompleteness creates room for innovation and hence this element symbolizes our faith in the student community to push the ambits of existing boundaries. If your society is not listed here, please let us know via the Slack link below.</p>
+				</Tooltip>
+			</>;
+		case "undiscovered":
+			const { name, symbol } = getIUPACTemporaryNameAndSymbol(props.index);
+
+			return <>
+				<div data-tooltip-id={`${props.index.toString()}-${symbol}`} data-tooltip-delay-hide={0}>
+					<GridItem className={`society undiscovered-soc block-${props.block.toLowerCase()}`} >
+						<span className="index">{props.index}</span>
+						<span className="year">?</span>
+						<span className="symbol">{symbol}</span>
+						<span className="size">NA</span>
+					</GridItem>
+				</div>
+				<Tooltip className="soc-tooltip" id={`${props.index.toString()}-${symbol}`} delayHide={0} delayShow={0}>
+					<span className="soc-tooltip-name">{name} (Not Discovered)</span>
+					<p className="soc-toolip-desc">This society has not been discovered yet and has been assigned a temporary name.</p>
+				</Tooltip>
+			</>;
+	}
+}
+
+function TableCell(props: TableCellProps) {
+    Modal.setAppElement('#root');
+    const [modalIsOpen, setIsOpen] = useState(false);	
+    
+    function openModal() {
+        if(!modalIsOpen) setIsOpen(true);
+    }
+    
+    
+    function closeModal() {
+        if(modalIsOpen) setIsOpen(false);
+    }
+
+    
+    return <div onClick={openModal}>
```

Comment:
Make this a `button` element or add a hover effect to change the cursor type to the clickable version.